src/main/java/edu/njit/jerse/automation/RepositoryAutomationEngine.java

```diff
@@ -0,0 +1,239 @@
+        if (Files.notExists(repoPath)) {
+            GitUtils.cloneRepository(repoUrl, branch, repoPath.toFile());
+        } else {
+            GitUtils.fetchRepository(repoPath.toFile());
+        }
+        return repoPath;
+    }
+
+    /**
+     * Processes a given repository by scanning for Java project roots. For each Java project found,
+     * this method applies the {@code AsheAutomation} process. The method supports processing
+     * multiple projects within a single repository.
+     *
+     * @param repoPath path to the local repository
+     * @param branch   the branch of the repository to be processed
+     * @throws IOException if an I/O error occurs while accessing the file system
+     */
+    private static void processSingleRepository(Path repoPath, String branch) throws IOException {
+        LOGGER.info("Processing repository at: {} for branch: {}", repoPath, branch);
+
+        List<File> projectRoots = ProjectRootFinder.findJavaRoots(repoPath.toFile());
+        for (File projectRoot : projectRoots) {
+            LOGGER.info("Processing project root: " + projectRoot.getPath());
+            AsheAutomation.iterateJavaFiles(projectRoot, projectRoot.getAbsolutePath());
+        }
+
+        LOGGER.info("Completed processing for branch: {} in repository path: {}", branch, repoPath);
+    }
+
+    /**
+     * Extracts the repository name from its URL.
+     *
+     * @param repoUrl URL of the repository
+     * @return name of the repository extracted from the URL
+     * @throws IllegalArgumentException if the repository URL is invalid or the name cannot be extracted
+     */
+    private static String extractRepoName(String repoUrl) {
+        LOGGER.info("Extracting repository name from URL: {}", repoUrl);
+
+        int lastSlashIndex = repoUrl.lastIndexOf('/');
+        int gitIndex = repoUrl.lastIndexOf(".git");
+
+        // Checks if ".git" is present in the URL and not at the end
+        boolean hasGitExtension = gitIndex > 0;
+
+        // Checks if there are characters after the last slash
+        boolean hasCharsAfterLastSlash = lastSlashIndex < repoUrl.length() - 1;
+
+        // Has '.git' extension. Example: https://github.com/jonathan-m-phillips/ASHE_Automated-Software-Hardening-for-Entrypoints.git
+        if (hasGitExtension && hasCharsAfterLastSlash) {
+            return repoUrl.substring(lastSlashIndex + 1, gitIndex);
+        }
+
+        // No '.git' extension, but a valid URL. Example: https://github.com/jonathan-m-phillips/ASHE_Automated-Software-Hardening-for-Entrypoints
+        if (hasCharsAfterLastSlash) {
+            return repoUrl.substring(lastSlashIndex + 1);
+        }
+
+        throw new IllegalArgumentException("Invalid repository URL");
+    }
+
+    /**
+     * The entry point of the application when trying to read and clone repositories from a CSV file.
+     * Expects two command-line arguments: the path to the repositories CSV file and the directory where
```

Comment:
Remove this sentence.  It is repetitive with the `@param` tag, and duplication leads to the possibility of inconsistency.